csrc/ops/alias.cpp

```diff
@@ -246,94 +265,45 @@ TensorView* squeeze(TensorView* x, const std::vector<bool>& to_squeeze) {
   return out;
 }
 
-TensorView* squeeze(TensorView* x, const std::vector<int64_t>& sizes) {
-  NVF_ERROR(x != nullptr, "Input is invalid.");
-  const auto ndims = static_cast<int>(x->domain()->noReductions().size());
-
-  NVF_ERROR(
-      ndims == int(sizes.size()),
-      "Invalid sizes for squeeze: ",
-      sizes,
-      ". Input tensor: ",
-      x->toString());
-
-  std::vector<bool> to_squeeze(ndims);
-  for (const auto idx : c10::irange(sizes.size())) {
-    to_squeeze[idx] = (sizes[idx] == 1);
-  }
-  return squeeze(x, to_squeeze);
-}
-
-TensorView* squeeze(TensorView* x, const std::vector<int64_t>& sizes, int dim) {
-  NVF_ERROR(x != nullptr, "Input is invalid.");
-  const auto ndims = static_cast<int>(x->domain()->noReductions().size());
-
-  NVF_ERROR(
-      ndims == int(sizes.size()),
-      "Invalid sizes for squeeze: ",
-      sizes,
-      ". Input tensor: ",
-      x->toString());
-
-  if (dim < 0) {
-    dim = ndims + dim;
-  }
-
-  NVF_ERROR(
-      dim >= 0 && dim < ndims,
-      "Invalid position to squeeze: ",
-      dim,
-      ". Input tensor: ",
-      x->toString());
-
-  if (sizes[dim] == 1) {
-    std::vector<bool> to_squeeze(ndims, false);
-    to_squeeze[dim] = true;
-    return squeeze(x, to_squeeze);
-  } else {
-    return set(x);
-  }
-}
-
-TensorView* squeeze(
-    TensorView* x,
-    const std::vector<int64_t>& sizes,
-    const std::vector<int64_t>& dims) {
+TensorView* squeeze(TensorView* x, const std::vector<bool>& to_squeeze) {
   NVF_ERROR(x != nullptr, "Input is invalid.");
-  const auto ndims = static_cast<int>(x->domain()->noReductions().size());
+  auto x_dom = x->domain()->noReductions();
+  const auto ndims = static_cast<int>(x_dom.size());
 
   NVF_ERROR(
-      ndims == int(sizes.size()),
-      "Invalid sizes for squeeze: ",
-      sizes,
+      ndims == (int)to_squeeze.size(),
+      "Invalid to_squeeze for squeeze: ",
+      to_squeeze,
       ". Input tensor: ",
       x->toString());
 
-  bool is_all_singleton_dimensions = true;
-
-  std::vector<bool> to_squeeze(ndims);
-  for (auto dim : dims) {
-    if (dim < 0) {
-      dim = ndims + dim;
+  std::vector<IterDomain*> out_domain;
+  for (const auto idx : c10::irange(ndims)) {
+    auto id = x_dom[idx];
+    if (to_squeeze[idx]) {
+      if (!id->isSymbolic()) {
+        NVF_CHECK(
+            id->isBroadcast(),
+            "Can not squeeze non-broadcasting dimension(s).");
+        NVF_CHECK(
+            !id->hasExpandedExtent(), "Can not squeeze expanded dimension(s).");
+        NVF_CHECK(
+            id->extent()->isConstScalar() && id->extent()->evaluate() == 1,
+            "Can not squeeze dimension(s) with size != 1.");
```

Comment:
Could we just skip the squeeze in these conditions? Then we could ignore trivial squeezes in the python frontend.

Reply:
The Python Frontend calls the other `squeeze` that skips non-broadcasted dimensions that are selected to be squeezed.